index.bs

```diff
@@ -3459,7 +3725,114 @@ The {{MLInstanceNormalizationOptions}} members are:
+
+    : <dfn>axes</dfn>
+    ::  
+        A sequence of {{unsigned long}}. The indices to the input dimensions to reduce. When this member is not present, it is assumed to be [1,2,3] that is, the reduction for the mean and variance values are calculated across all the input features for each individual sample in the batch.
+
+    : <dfn>epsilon</dfn>
+    ::
+        A {{float}} scalar. Specifies a small value to prevent computational error due to divide-by-zero.
+</dl>
+
+<div>
+    **Arguments:**
+        - *input*: an {{MLOperand}}. The input 4-D tensor.
+        - *options*: an optional {{MLLayerNormalizationOptions}}. The optional parameters of the operation.
+
+    **Returns:** an {{MLOperand}}. The layer-normalized 4-D tensor of the same shape as *input*.
+</div>
+
+<details open algorithm>
+  <summary>
+    The <dfn method for=MLGraphBuilder>layerNormalization(|input|, |options|)</dfn> method steps are:
+  </summary>
+  <div class=algorithm-steps>
+    1. [=Assert=]: the type of |input| is {{MLOperand}}.
+    1. [=Assert=]:  the type of |options|.{{MLLayerNormalizationOptions/scale}} is {{MLOperand}}.
+    1. If the [=rank=] of |options|.{{MLLayerNormalizationOptions/scale}} is not equal to the [=list/size=] of |options|.{{MLLayerNormalizationOptions/axes}}, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
+    1. [=Assert=]: the type of |options|.{{MLLayerNormalizationOptions/bias}} is  {{MLOperand}}.
+    1. If the [=rank=] of |options|.{{MLLayerNormalizationOptions/bias}} is not equal to the [=list/size=] of |options|.{{MLLayerNormalizationOptions/axes}}, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
+    1. [=map/For each=] |index| in [=the range=] 0 to the [=list/size=] of |options|.{{MLLayerNormalizationOptions/axes}, exclusive:
+        1. Let |axis| be |options|.{{MLLayerNormalizationOptions/axes}}[|index|].
+        1. If |axis| is greater or equal to the [=list/size=] of |input|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
+        1. Let |size| be |input|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}[|axis|].
+        1. If |options|.{{MLLayerNormalizationOptions/scale}}.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}[|index] is not equal to |size|, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
+        1. If |options|.{{MLLayerNormalizationOptions/bias}}.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}[|index] is not equal to |size|, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
```

Comment:
GitHub's add a suggestion UI (Ctrl+g) is usually handy for smaller changes. It turns the suggestions into commits that can be incorporated by the author:
```suggestion
        1. If |options|.{{MLLayerNormalizationOptions/scale}}.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}[|index|] is not equal to |size|, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
        1. If |options|.{{MLLayerNormalizationOptions/bias}}.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}[|index|] is not equal to |size|, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
```

Reply:
Fixed.

Reply:
Fixed.

Reply:
Fixed in https://github.com/webmachinelearning/webnn/pull/478/commits/2d976cb272c77cc4adc81e7405959dec011b8ae3